File: Net2.0/Examples/Misc/VB/SuperAddin/SuperAddin/HostApplication.vb
Imports LateBindingApi.Core
Imports System.ComponentModel
Imports Extensibility

Public Class HostApplication
    Implements IDisposable

#Region "Fields"

    Private disposedValue As Boolean = False        ' To detect redundant calls
    Private _hostApp As COMObject

#End Region

#Region "Properties"

    Public ReadOnly Property Application() As COMObject
        Get
            Return _hostApp
        End Get
    End Property

    Public ReadOnly Property ComponentName() As String
        Get
            Return TypeDescriptor.GetComponentName(_hostApp.UnderlyingObject)
        End Get
    End Property

    Public ReadOnly Property Name() As String
        Get
            Return _hostApp.UnderlyingTypeName
        End Get
    End Property

    Public ReadOnly Property Version() As String
        Get
            Return Invoker.PropertyGet(_hostApp.UnderlyingObject, "Version")
        End Get
    End Property

#End Region

#Region "Construction"

    Public Sub New(ByVal comProxy As Object, ByVal ConnectMode As ext_ConnectMode, ByVal AddInInst As Object, ByRef custom As Array)

        Dim typeComponent As String = System.ComponentModel.TypeDescriptor.GetComponentName(comProxy)
        Select Case typeComponent
            Case "Microsoft Excel"
                _hostApp = New NetOffice.ExcelApi.Application(Nothing, comProxy)
            Case "Excel"
                _hostApp = New NetOffice.ExcelApi.Application(Nothing, comProxy)


            Case "Microsoft Word"
                _hostApp = New NetOffice.WordApi.Application(Nothing, comProxy)
            Case "Word"
                _hostApp = New NetOffice.WordApi.Application(Nothing, comProxy)

            Case "Microsoft Outlook"
                _hostApp = New NetOffice.OutlookApi.Application(Nothing, comProxy)
            Case "Outlook"
                _hostApp = New NetOffice.OutlookApi.Application(Nothing, comProxy)

            Case "Microsoft PowerPoint"
                _hostApp = New NetOffice.PowerPointApi.Application(Nothing, comProxy)
            Case "PowerPoint"
                _hostApp = New NetOffice.PowerPointApi.Application(Nothing, comProxy)

            Case "Microsoft Access"
                _hostApp = New NetOffice.AccessApi.Application(Nothing, comProxy)
            Case "Access"
                _hostApp = New NetOffice.AccessApi.Application(Nothing, comProxy)
        End Select

    End Sub

#End Region

#Region " IDisposable Members "

    ' IDisposable
    Protected Overridable Sub Dispose(ByVal disposing As Boolean)
        If Not Me.disposedValue Then
            If disposing Then
                ' TODO: free other state (managed objects).
            End If

            ' TODO: free your own state (unmanaged objects).
            ' TODO: set large fields to null.
        End If
        Me.disposedValue = True
    End Sub


    ' This code added by Visual Basic to correctly implement the disposable pattern.
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(ByVal disposing As Boolean) above.
        Dispose(True)
        GC.SuppressFinalize(Me)
    End Sub

#End Region

End Class

File: Examples/Misc/C#/SuperAddin/HostApplication.cs
using System;
using Extensibility;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using System.ComponentModel;

using LateBindingApi.Core;
using Office = NetOffice.OfficeApi;
using Excel = NetOffice.ExcelApi;
using Word = NetOffice.WordApi;
using Outlook = NetOffice.OutlookApi;
using PowerPoint = NetOffice.PowerPointApi;
using Access = NetOffice.AccessApi;

namespace SuperAddinCSharp
{
    /// <summary>
    /// represents the office host application
    /// </summary>
    public class HostApplication : IDisposable
    {
        #region Fields

        COMObject                 _hostApp;

        #endregion

        #region Properties

        /// <summary>
        /// gets office host application object
        /// </summary>
        public COMObject Application
        {
            get 
            {
                return _hostApp;
            }
        }
        
        /// <summary>
        /// returns component host name of host app
        /// </summary>
        public string ComponentName
        {
            get
            {
                return TypeDescriptor.GetComponentName(_hostApp.UnderlyingObject);
            }
        }

        /// <summary>
        /// returns name of host app
        /// </summary>
        public string Name
        {
            get
            {
               return _hostApp.UnderlyingTypeName;
            }
        }

        /// <summary>
        /// returns version of host app
        /// </summary>
        public string Version
        {
            get 
            {
                return (string)Invoker.PropertyGet(_hostApp.UnderlyingObject, "Version");
            }
        }

        #endregion

        #region Construction

        public HostApplication(object comProxy, ext_ConnectMode ConnectMode, object AddInInst, ref Array custom)
        {
            string typeComponent = System.ComponentModel.TypeDescriptor.GetComponentName(comProxy);
            switch (typeComponent)
            { 
                case "Microsoft Excel":
                case "Excel":
                    _hostApp = new Excel.Application(null, comProxy);
                    break;
                case "Microsoft Word":
                case "Word":
                    _hostApp = new Word.Application(null, comProxy);
                    break;
                case "Microsoft Outlook":
                case "Outlook":
                    _hostApp = new Outlook.Application(null, comProxy);
                    break;
                case "Microsoft PowerPoint":
                case "PowerPoint":
                    _hostApp = new PowerPoint.Application(null, comProxy);
                    break;
                case "Microsoft Access":
                case "Access":
                    _hostApp = new Access.Application(null, comProxy);
                    break;
            }
        }

        #endregion

        #region IDisposable Members

        public void Dispose()
        {
            if (null != _hostApp)
                _hostApp.Dispose();
        }

        #endregion
    }
}
